src/sentry/web/frontend/react_page.py

```diff
@@ -107,12 +106,20 @@ def handle_auth_required(self, request: Request, *args, **kwargs) -> HttpRespons
         # For normal users, let parent class handle (e.g. redirect to login page)
         return super().handle_auth_required(request, *args, **kwargs)
 
-    def handle(self, request: Request, organization, **kwargs) -> HttpResponse:
-        if "project_id" in kwargs and request.GET.get("onboarding"):
-            project = Project.objects.filter(
-                organization=organization, slug=kwargs["project_id"]
-            ).first()
-            first_event_pending.send(project=project, user=request.user, sender=self)
+    def handle(self, request: Request, organization: RpcOrganization, **kwargs) -> HttpResponse:
+        if "project_slug" in kwargs and request.GET.get("onboarding"):
+            project = next(
+                (p for p in organization.projects if p.slug == kwargs["project_slug"]), None
+            )
```

Comment:
This looks like it could be expensive. Having RpcOrganization inline all of the projects could be a huge payload as some orgs have 1000s of projects. Could we adapt the signal to pass the project slug instead? The project slug is unlikely to shift under us during onboarding and even if it does the impact of that is very low.